src/Tizen.NUI/src/public/BaseComponents/VectorGraphics/DrawableGroup.cs

```diff
@@ -0,0 +1,71 @@
+/* 
+* Copyright(c) 2021 Samsung Electronics Co., Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*/
+
+using System;
+using System.ComponentModel;
+
+namespace Tizen.NUI.BaseComponents.VectorGraphics
+{
+    /// <summary>
+    /// A class enabling to hold many Drawable objects. As a whole they can be transformed, their transparency can be changed.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class DrawableGroup : Drawable
+    {
+        /// <summary>
+        /// Creates an initialized DrawableGroup.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public DrawableGroup() : this(Interop.DrawableGroup.New(), true)
+        {
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+        }
+
+        internal DrawableGroup(global::System.IntPtr cPtr, bool cMemoryOwn) : base(cPtr, cMemoryOwn)
+        {
+        }
+
+        /// <summary>
+        /// Add drawable object to the DrawableGroup. This method is similar to registration.
+        /// </summary>
+        /// <param name="drawable">Drawable object</param>
+        /// <exception cref="ArgumentNullException"> Thrown when drawable is null. </exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void AddDrawable(Drawable drawable)
```

Comment:
I think, you do better to hold drawable object in this class to prevent released by GC, 
In general, objects belonging to somewhere should not be removed by GC. 
And this responsibility is placed on the container object.
It is similar to View and child relationship

So, I recommend to hold drawable object in DrawableGroup

Reply:
Can you tell me in more detail about how to hold the drawable object you are talking about.
In NUI, even if the view transfers ownership to the container as you say, 
if the user does not explicitly call dispose(), a memory error occurs.
I know, Calling Dispose() is different from freeing the memory,
but I don't think it's different that the user is responsible for controlling the object until the end.
I think, If the DrawbleGroup is released before the added Drawable object, it is safe to leave it to the GC as long as it does not cause internal memory problems.

And DrawableGroup and Drawble don't inherit View.
So I don't know if Drawble and DrawbleGroup should follow those rule like View and Child relationship.

I need more study of C#.
Please tell me how to hold this when the GC wants to release the child of the container.

Reply:
우선, 객체를 `hold`하는 것은 컨테이너 객체 내부에 `drawable`객체에 대한 참조를 유지 하고 있는 것입니다.(예를 들어 내부적으로 List<Drawable> 객체를 가지고 있어서 추가되는 모든 Drawable을 List에 넣어주는것으로 참조 유지를 할 수 있습니다.)

1. DrawableGroup객체는 drawable객체를 contain하고 있는 객체이기 때문에 contain하고 있는 동안에는 drawable객체는 GC에 의해서 삭제 되면 안됩니다.
2. Disposable과 상관 없이, Drawble객체는 (아마도) 소멸자에서 Native객체를 정리하기 때문에, GC에 의해서 managed객체 (Drawable객체)가 삭제될때, Native객체도 삭제 될것입니다.
3. 이렇게 된 경우에 DrawableGroup객체는 자신이 contain하고 있는 객체에 접근하는것이 문제가 될것입니다 (dangling object access)
그래서 지금과 같은 구현을 제공하는 경우, API사용자는 DrawableGroup객체 뿐만 아니라, DrawableGroup에 포함되는 모든 객체에 대한 참조를 어디에선가 유지하고 있어야 합니다.
그런데, 이렇게 API를 사용하는것은 매우 불편한 것입니다.
당장 CanvasViewSamsple.cs 샘플 코드만 보더라도 모든 객체를 field로 저장하고 있는 것을 알 수 있고, 이는 매우 불편합니다.

View의 예를 들었는데, DrawableGroup이 View라는 의미가 아니였고, View가 Container역할을 하고 있고, 일반적인 Container가 해야할 일을 하고 있으니 그걸 보고 비슷하게 구현하라는 의미였습니다.

즉, C#에서는 Container객체에 넣은 child 객체에 대해서는 개발자가 따로 래퍼런스 관리를 하지 않더라도 GC에 의해서 Release되면 안됩니다.

Reply:
상세한 설명 감사합니다. dali쪽에서도 비슷한 문제가 있는데 다른부분들도 리뷰 주신 방향으로 고치는게 맞을것 같습니다.
리뷰 주신대로 DrawableGroup에 참조를 유지 할 수 있는 List를 추가 했습니다.
이렇게 변경하는게 맞으면 CanvasView의 AddDrawable도 동일하게 수정하도록 하겠습니다.
다시 한번 검토 부탁드립니다.